synthetic-monitoring/sm.libsonnet

```diff
@@ -0,0 +1,102 @@
+local sm = import 'sm.libsonnet';
+{
+  syntheticMonitoring+:: {
+    grafanaCheck: sm.new('grafana-com', 'https://grafana.com/')
+                  + sm.withHttp()
+                  + sm.withProbes('all'),
+  },
+}
+{
+  new(name, target):: {
+    frequency: 60000,
+    offset: 0,
+    timeout: 2500,
+    enabled: true,
+    labels: [],
+    target: target,
+    job: name,
+  },
+
+  withHttp():: {
+    settings: {
```

Comment:
```suggestion
    settings+: {
```

Reply:
Wouldn't that make more sense?

Reply:
Actually, I believe it is one type per check, I should confirm that. In which case `newHttp` etc would make more sense.

Reply:
If that is the case, nesting it in `settings.http` looks a bit superfluous.

Reply:
exactly.

Reply:
@Duologic I've updated the API to account for this. It is also now slightly more concise. Work for you?